Understanding IPV4 and IPV6 protocols (#2186)
section-engineering-education/engineering-education: articles/internet-protocols/index.md

---
layout: engineering-education
status: publish
published: true
url: /engineering-education/internet-protocols/
title: Understanding IPV4 and IPV6 protocols
description: This article will be a walk through the IPv4 and IPv6 Internet protocols which make communication on the internet possible. 
author: mackrine-awino
date: 2021-05-05T00:00:00-13:00
topics: [Protocols]
excerpt_separator: <!--more-->
images:

  - url: /engineering-education/internet-protocols/hero.jpg
    alt: Internet Protocols Hero Image
---
Internet protocols are what make communication on the internet possible. Every device connected to the internet is assigned an IP address. This enables the identification and location of networked computers on the internet. 
<!--more-->

> Internet protocols also enable the transportation of data items on the internet. Datagrams are transmitted from one device to another without connections to one another by using internet protocols. The datagrams are transmitted with utmost independence from all others.

Reviewer: Briefly introducing what are datagrams before explaining how they are transmitted from one device to another

### Table of contents

1. [Prerequisites](#prerequisites)
2. [Introduction to IPv4 and IPv6](#introduction-to-IPv4-and-IPv6)
3. [Internet protocol version 4](#internet-protocol-version-4)
4. [Internet protocol version 6](#internet-protocol-version-6)
5. [Similarities between IPv4 and IPv6](#similarities-between-IPv4-and-IPv6)
6. [Differences between IPv4 and IPv6](#differences-between-IPv4-and-IPv6)
7. [Why does IPv6 take preference to IPv4?](#why-does-IPv6-take-preference-to-IPv4)
8. [Conclusion](#conclusion)

### Prerequisites

To understand the content of this article, a reader should have the following:
- A beginner's understanding of computer networking.

- Basic knowledge of internet communication and data transmission.

### Introduction to IPv4 and IPv6

IPv4 and IPv6 are the actual protocols tasked with data transmission in the form of packets and datagrams. 

IPv4 protocol is mainly used with [ethernets](https://en.wikipedia.org/wiki/Ethernet)	during packet switching in the link-layer networks.	

> 
> IPv6 protocol being more newfangled has improved capabilities as compared to IPV4. It can even have infinite addresses hence is replacing IPV4 due to the daily growth in technology increase number of networks being set up frequently.

Reviewer: ```suggestion
IPv6 protocol being more newfangled has improved capabilities as compared to IPV4. Hence, it can even have infinite addresses replacing IPV4 due to the daily growth in technology and increases the number of networks being set up frequently.
```

![IPv4 Communication Protocol](Screenshot_20210428-104617.png)

IPv4 still controls the internet traffics despite the development of IPv6 using a 32bit address space providing over four million unique addresses.

IPv4 being a connectionless protocol, does not guarantee if data is delivered, neither does it arrange the data packets properly. Some packets may also have duplicates, so the organization is all addressed by the layer that handles transport, i.e., [TCP](https://www.sdxcentral.com/resources/glossary/transmission-control-protocol-tcp/#:~:text=Transmission%20Control%20Protocol%20(TCP)%20%E2%80%93,referred%20to%20as%20TCP%2FIP.)or the [UDP](https://en.wikipedia.org/wiki/User_Datagram_Protocol). IPv4 also has special and multicast addresses for [private](https://en.wikipedia.org/wiki/Private_network) networks with approximately eighteen million addresses.

IPv4 address digits are separated with decimal points as shown below:

#### Examples of IPv4 addresses

               172.16.254.1

               169.254.255.255 
### IPv4 addressing

Three different types of addressing modes supported in IPv4, namely; 
- Unicast addressing mode
- Broadcast addressing mode

> - Multicast addressing mode

Reviewer: Briefly explain each of these addressing modes

                           

### Internet protocol version 6

> Internet protocol version 6 is the newest version of the internet protocol suite as of now developed to replace the fourth version, i.e., the IPv4. It was brought into existence by the [IETF](https://www.ietf.org/blog/ipv6-internet-standard/) because IPv4 has exhausted its addresses. IPv6 was intended to replace IPv4. However, this has not been the case; IPv4 has continued to live on.

Reviewer: ```suggestion
Internet protocol version 6 is the newest version of the internet protocol suite as of now developed to replace the fourth version, i.e., the IPv4. IPv6 was brought into existence by the [IETF](https://www.ietf.org/blog/ipv6-internet-standard/) because IPv4 had exhausted its addresses. IPv6 was intended to replace IPv4. However, this has not been the case; IPv4 has continued to live on.

4. Both IPv4 and IPv6 allow for the manual assignment of IP addresses.

### Differences between IPv4 and IPv6

1. IPv6 has a packet flow identification in the header while IPv4 offers no packet flow information
2. IPv6 uses 128-bit hexadecimal, i.e., base 16 IP addresses, while IPv4 uses 32-bit  IP addresses written in a decimal number system which is a base 10.
3. IPv4 supports dynamic host configuration protocol setup, while IPv6 supports renumbering address setup.
4. IPv4 has not connection integrity, whereas IPv6 has end-to-end connection integrity.
5. IPv4 uses the decimal representation of addresses, while IPv6 uses hexadecimal representation.

### Why does IPv6 take preference to IPv4?

Other than having larger addressing space, IPv6 has other beneficial features ad discussed below;
- It allows for the address allocated to be done in a hierarchy enhancing supernetting.
- It increases the [multicast](https://en.wikipedia.org/wiki/Multicast_address#:~:text=From%20Wikipedia%2C%20the%20free%20encyclopedia,for%20a%20designated%20network%20service.) addressing hence adjustment of service delivery.
- IPv6 ensures the portability and security of its host devices.

 ### Conclusion

> 
> TThe rapidly growing population of the world is pushing for the gradual phase-out to Ipv4. This, together with the frequent advancement in technology in almost every sector in the community, leads to the development of more networks; hence the depletion of all IPv4 addresses IPv6 is slowly becoming the future of the internet IP addressing system.	
> 

Reviewer: ```suggestion
The rapidly growing population of the world is pushing for the gradual phase-out to Ipv4. In addition, the frequent advancement in technology in almost every sector in the community leads to the development of more networks; hence the depletion of all IPv4 addresses IPv6 is slowly becoming the future of the internet IP addressing system.
```
